consumer_pipes.py

```diff
@@ -0,0 +1,263 @@
+
+        return x
+
+    def get_feature_names(self):
+
+        self.feature_names
+
+
+
+class DataFrameImputer(BaseEstimator,TransformerMixin):
+    
+    def __init__(self):
+        
+        
+        self.feature_names=[]
+        self.impute_dict={}
+        
+    
+    def fit(self,x,y=None):
+        
+        self.feature_names=x.columns
+        
+        for col in x.columns:
+            
+            if x[col].dtype=='O':
+                self.impute_dict[col]='missing'
+            else :
+                self.impute_dict[col]=x[col].median()
+                
+        return self
+    
+    def transform(self,x) :
+        
+        return x.fillna(self.impute_dict)
+        
+    def get_feature_names(self) :
+                
+        return self.feature_names
+
+class string_clean(BaseEstimator,TransformerMixin):
+    
+    def __init__(self,replace_it,replace_with):
+        
+        
+        self.feature_names=[]
+        self.replace_it=replace_it
+        self.replace_with=replace_with        
+    
+    def fit(self,x,y=None):
+
+        self.feature_names=x.columns
+
+        return self 
+    
+    def transform(self,x) :
+        
+        for col in x.columns:
+
+            x[col]=x[col].str.replace(self.replace_it,self.replace_with)
+        
+        return x
+
+    def get_feature_names(self) :
+                
+        return self.feature_names
+
+
+class creat_dummies(BaseEstimator,TransformerMixin):
+    
+    def __init__(self,freq_cutoff=0):
+        
+        
+        self.freq_cutoff=freq_cutoff
+        self.var_cat_dict={}
+        self.feature_names=[]
+        
+    
+    def fit(self,x,y=None):
+
+        for col in x.columns:
+
+            k=x[col].value_counts()
+            cats=k.index[k>=self.freq_cutoff][:-1]
+
+            self.var_cat_dict[col]=list(cats)
+
+        for col in self.var_cat_dict.keys():
+
+            for cat in self.var_cat_dict[col]:
+                self.feature_names.append(col+'_'+str(cat))
+
+        return self 
+    
+    def transform(self,x) :
+        
+        dummy_data=x.copy()
+
+        for col in self.var_cat_dict.keys():
+
+            for cat in self.var_cat_dict[col]:
+
+                name=col+'_'+str(cat)
+
+                dummy_data[name]=(dummy_data[col]==cat).astype(int)
+
+            del dummy_data[col]
+
+        return dummy_data
+
+    def get_feature_names(self) :
+                
+        return self.feature_names
+
+    
+class tfidf(BaseEstimator,TransformerMixin):
+    
+    def __init__(self):
+        
+        self.feature_names=[]
+        
+        vectorizer = TfidfVectorizer(analyzer='word',stop_words='english',max_df=0.8,min_df=0.01,max_features=200)
+    
+    def fit(self,x,y=None):
+
+        self.feature_names=x.columns
+        
+        return self 
+    
+    def transform(self,x) :
+              
+        for col in x.columns:
+            corpus = col
+            df = vectorizer.fit_transform(corpus)

```

Comment:
using fit_transform here will fit on every new data you transform which defies the purpose of having training data and in general is not a right thing to do . you should fit in function `fit` and transform in `transform`